### PR TITLE
Updated examples and fixes

### PR DESCRIPTION
**Pull request by Surax98**
# Summary

The Kubeconfig wasn't correctly mounted in the kubernetes deployment and the Ping function tried to create a clientset. Now They should be fixed.
The kubeconfig is not generated and sent anymore if the KUBECONFIG env is set on the InterLink host
ConfigMaps/Secrets retrieving now re-works and every file related to a specific job (for the Slurm plugin) is now stored in the same subdirectory in the form of .local/interlink/jobs/namespace-podUID (this includes JobID, job.sh, output and every configmap/secret retrieved from the pod).
Also, now InterLink queries sidecars for running/pending pods only. If the VK asks the status for succeeded/terminated pods, InterLink will provide a cached status

**Related issue :**
#130 #25 #90 #134

### CODE REVIEW

File: examples/interlink-slurm/README.md [hunk 1]
```diff
@@ -28,7 +28,28 @@ minikube start --kubernetes-version=1.24.3
...
+```bash
+export PATH_TO_KUBECONFIG=$HOME/.kube/config
+export CA_DATA=$(cat $HOME/.minikube/ca.crt | base64 -w0)
+export CERT_DATA=$(cat $HOME/.minikube/profiles/minikube/client.crt | base64 -w0)
+export KEY_DATA=$(cat $HOME/.minikube/profiles/minikube/client.key | base64 -w0)
+
+mkdir -p interlink/config
+
+cp $PATH_TO_KUBECONFIG interlink/config/kubeconfig.yaml 
+cp $PATH_TO_KUBECONFIG vk/kubeconfig.yaml
```
Comment (dciangot) [inline examples/interlink-slurm/README.md:44]: we do not need this anymore right? @Surax98

File: pkg/common/func.go [hunk 2]
```diff
@@ -244,18 +139,67 @@ func PingInterLink(ctx context.Context) (error, bool) {
...
+		return nil, false, retVal
+	}
+}
+
+func CreateClientsetFrom(ctx context.Context, body string) error {
+	counter := 0
+	for {
+		var returnValue, _ = json.Marshal("Error")
+		reader := bytes.NewReader([]byte(body))
+		req, err := http.NewRequest(http.MethodPost, InterLinkConfigInst.Interlinkurl+":"+InterLinkConfigInst.Interlinkport+"/setKubeCFG", reader)
```
Comment (dciangot) [inline pkg/common/func.go]: do we need really a clientset in interlink now? if yes for what? @Surax98

File: examples/interlink-slurm/README.md (hunk 1 above)
Comment (Surax98) [reply examples/interlink-slurm/README.md:44]: Correct, not needed anymore

File: pkg/common/func.go (hunk 2 above)
Comment (Surax98) [reply pkg/common/func.go]: Removed. There only were a couple of previously used functions that have been removed

**Pull request merged**